L1Trigger/L1THGCal/plugins/be_algorithms/SingleCellClusterAlgo.cc

```diff
@@ -32,14 +35,24 @@ class SingleCellClusterAlgo : public Algorithm<HGCalTriggerCellBestChoiceCodec>
         }
 
     private:
+        
+        edm::ParameterSet geometry_config_;
+        std::string geometry_name_;
         std::unique_ptr<l1t::HGCalTriggerCellBxCollection> cluster_product_;
+        edm::ESHandle<HGCalGeometry> hgceeGeoHandle_;
+        edm::ESHandle<HGCalGeometry> hgchefGeoHandle_;
+        HGCalTriggerCellCalibration calibration_;    
 
 };
 
 /*****************************************************************/
-void SingleCellClusterAlgo::run(const l1t::HGCFETriggerDigiCollection& coll) 
+void SingleCellClusterAlgo::run(const l1t::HGCFETriggerDigiCollection& coll, const edm::EventSetup& es) 
 /*****************************************************************/
 {
+    HGCalTriggerGeometryBase::es_info info;
+    es.get<IdealGeometryRecord>().get("HGCalEESensitive", info.topo_ee);
+    es.get<IdealGeometryRecord>().get("HGCalHESiliconSensitive", info.topo_fh);
```

Comment:
* Remove the hardcoded names and use config parameters instead
* Don't use the trigger geometry es_info, use the topology handles defined as members of the class